File: part_1/hw4/ex_A_3_3_g.m
cvx_begin
    variable x;
    variable y;
    minimize(norm(x + y + 1));
    subject to
        pow_p(x, 3) + pow_p(y, 3) <= 1;
        x >= 0;
        y >= 0;
cvx_end

fprintf('status:'); 
disp(cvx_status);

fprintf('optimal value:'); 
disp(cvx_optval )

fprintf('optimal x:\n'); 
disp(x)

fprintf('optimal y:\n'); 
disp(y)

File: part_1/hw4/ex_A_3_3_g.py
import cvxpy as cp

# Create two scalar optimization variables.
x = cp.Variable()
y = cp.Variable()

# Create two constraints.
constraints = [cp.pos(x) ** 3 + cp.pos(y) ** 3 <= 1, x >= 0, y >= 0]

# Form objective.
obj = cp.Minimize((x - y + 2)**2)

# Form and solve problem.
prob = cp.Problem(obj, constraints)
prob.solve()  # Returns the optimal value.
print("status:", prob.status)
print("optimal value", prob.value)
print("optimal var", x.value, y.value)

